src/DotNetWorker/Definition/GrpcFunctionMetadata.cs

```diff
@@ -14,6 +17,13 @@ public GrpcFunctionMetadata(FunctionLoadRequest loadRequest)
             Name = loadRequest.Metadata.Name;
             PathToAssembly = Path.GetFullPath(loadRequest.Metadata.ScriptFile);
             FunctionId = loadRequest.FunctionId;
+
+            var grpcbindings = loadRequest.Metadata.Bindings;
+            var grpcInputBindings = grpcbindings.Where(kv => kv.Value.Direction == BindingInfo.Types.Direction.In);
+            var grpcOutputBindings = grpcbindings.Where(kv => kv.Value.Direction != BindingInfo.Types.Direction.In);
+
+            InputBindings = grpcInputBindings.ToImmutableDictionary(kv => kv.Key, kv => new GrpcBindingMetadata(kv.Value) as BindingMetadata);
```

Comment:
Can you not specify the return type and avoid the explicit cast?

Reply:
I think because it's a lambda, I'd have to do -- `new Func<KeyValuePair<string, BindingInfo>, BindingMetadata>(kv => new GrpcBindingMetadata(kv.Value));` which I can do above and re-use for both.